tinygrad/tensor.py

```diff
@@ -341,10 +341,10 @@ def repeat(self, repeats):
 
   # TODO: make this nicer with syntactic sugar in slice
   def chunk(self, num, dim):
-    slice_params = [[(0, s) for s in self.shape] for _ in range(num)]
-    for i,k in enumerate(range(0, self.shape[dim], self.shape[dim]//num)):
-      slice_params[i][dim] = (k, min(self.shape[dim], k+self.shape[dim]//num))
-    return [self.slice(p) for p in slice_params]
+    slice_params = [[slice(None) for s in self.shape] for _ in range(ceil(self.shape[dim]/ceil(self.shape[dim]/num)))]
```

Comment:
Do you need to initialize with `slice(None)` here? does initializing with just `None` work?

Reply:
Replacing slice(None) with None breaks, my idea with the slice(None) is ":" but i think None acts different as it throws `AssertionError: zeros not allowed in shape (1, 0, 13)`

Btw, I have plans to replace the slice with .pad in .pad2d as well should I do it? Maybe I can combine them in this PR as well as it will be small

Reply:
Should keep that in a separate PR. I'll just merge this one now.